common/_static/helpfulness.js

```diff
@@ -0,0 +1,345 @@
+			show: function ()
+			{
+				if (!this.closed && !this.completed && this.shouldShow && !this.$container.is(":visible"))
+				{
+					var documentHeight = $(document).height();
+					var windowHeight = $(window).height();
+					var enoughScrollbars = (documentHeight - windowHeight) > 200;
+
+					if ($(window).scrollTop() > 100 || !enoughScrollbars)
+					{
+						this.$container.slideDown('fast');
+					}
+				}
+			},
+
+			submitYes: function()
+			{
+				if (this.targetKeyId)
+				{
+					$(".helpfulness-form").hide();
+					this.showMessage(".processing");
+
+					$.ajax({
+						context: this,
+						url: options.domain + "/base/HelpfulnessLibrary/SubmitYes",
+						type: "POST",
+						data: {
+							pickerId: this.targetPickerId,
+							contentId: this.targetDocumentId,
+							docId: this.targetDocId
+						},
+						success: function (data)
+						{
+							if (data != null && data == "success")
+							{
+								this.showMessage(".success");
+
+								this.setStorage(this.targetKeyId);
+								this.completed = true;
+
+								this.initLoad();
+							}
+							else
+							{
+								this.showMessage(".error");
+							}
+						},
+						error: function (error)
+						{
+							this.showMessage(".error");
+						}
+					});
+				}
+			},
+
+			submitNo: function (value)
+			{
+				if (this.targetKeyId)
+				{
+					$(".helpfulness-form-no").hide();
+					this.showMessage(".processing");
+
+					$.ajax({
+						context: this,
+						url: options.domain + "/base/HelpfulnessLibrary/SubmitNo",
+						type: "POST",
+						data: {
+							pickerId: this.targetPickerId,
+							contentId: this.targetDocumentId,
+							docId: this.targetDocId,
+							reason: value
+						},
+						success: function (data)
+						{
+							if (data != null && data == "success")
+							{
+								this.showMessage(".success");
+
+								this.setStorage(this.targetKeyId);
+								this.completed = true;
+
+								this.initLoad();
+							}
+							else
+							{
+								this.showMessage(".error");
+							}
+						},
+						error: function (error)
+						{
+							this.showMessage(".error");
+						}
+					});
+				}
+			},
+
+			getStorage: function (contentId)
+			{
+				if (window.localStorage)
+				{
+					return localStorage.getItem("helpfulness-" + contentId);
+				}
+
+				return null;
+			},
+
+			setStorage: function (contentId)
+			{
+				if (window.localStorage)
+				{
+					try
+					{
+						localStorage.setItem("helpfulness-" + contentId, 1);
+						return true;
+					}
+					catch (e)
+					{
+						if (e.name === 'QUOTA_EXCEEDED_ERR' || e.name === 'NS_ERROR_DOM_QUOTA_REACHED')
+						{
+							return false;
+						}
+						else
+						{
+							return false;
+						}
+					}
+				}
+
+				return false;
+			},
+
+			showMessage: function (message)
+			{
+				this.$content.children(".messages").hide();
+				this.$content.children(message).show();
+			}
+		};
+	};
+})(jQuery);
+
+$(function ()
+{
+	$.Helpfulness({
+		maxCharacters: 160,
```

Comment:
I agree with wanting to avoid tech support questions, but I don't feel that 160 chars is enough to provide any meaningful feedback outside of "this is great" or "this is not-so-great".

In fact, the previous sentence I just typed is 185 characters, and hardly says anything useful, but wouldn't be allowed in the feedback box we're showing to users.